shared/test/police_box.rb

```diff
@@ -0,0 +1,50 @@
+require 'timecop'
+
+#
+# PoliceBox - Because sometimes tests are timey-wimey.
+#
+# This is a utility for tests that should use fake time using Timecop in some
+# circumstances (like when running in true isolation) and real time in other
+# circumstances (like when running against a real database).
+#
+class PoliceBox
+  def initialize(in_real_time)
+    @time_circuits_on = !in_real_time
+  end
+
+  #
+  # Create a time machine for the duration of a block, which may or may not
+  # actually time travel, automatically cleaned up when the block is done.
+  # @param [Boolean] in_real_time - Pass TRUE to disable time travel and
+  #   have the test actually wait for the real amount of time.
+  #   Think of this as the stay_under_88_mph option if it helps.
+  #
+  def self.in_box(in_real_time = false)
+    time_machine = PoliceBox.new(in_real_time)
+    time_machine.pause_time
+    yield time_machine
+  ensure
+    time_machine.resume_time
+  end
+
+  #
+  # Pass time in the test - using real or fake time, depending on how the
+  # time machine is configured.
+  # @param [Integer] seconds - How far to jump forward
+  #
+  def time_travel(seconds)
+    if @time_circuits_on
+      Timecop.travel seconds
+    else
+      sleep seconds
+    end
+  end
+
+  def pause_time
+    Timecop.freeze if @time_circuits_on
```

Comment:
Note that I'd expect this to `raise` if time circuits are off, except we can't if we want tests to be "ambidextrous". Probably worth documenting this (also with `resume_time`).